File: 2015/2/2.py

def parse_file(filename):
    input = []
    with open(filename, 'r') as file:
        for line in file:
            parts = line.strip().split('x')
            input.append(list(map(int, parts)))
    return input

def wrapping_papers(input):
    total =0
    for dims in input:
        wrapping = 2*dims[0]*dims[1] + 2*dims[0]*dims[2] + 2*dims[1]*dims[2]
        dims.sort()
        slack = dims[0]*dims[1]
        total += wrapping + slack
    return total

def ribbons(input):
    total = 0
    for dims in input:
        wrapping = dims[0]*dims[1]*dims[2]
        dims.sort()
        slack = 2*dims[0] + 2*dims[1]
        total += wrapping + slack
    return total


filename = "2.txt"
input = parse_file(filename)
ans1 = wrapping_papers(input)
ans2 = ribbons(input)
print("Answer for the part1: ", ans1)
print("Answer for the part2: ", ans2)
    

File: 2015/2/2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"slices"
)

func parseFile(filename string) [][]int {
	file, err := os.Open(filename)
	if err != nil {
		fmt.Printf("Failed to open file: %v", err)
	}
	defer file.Close()

	var result [][]int
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text();
		parts := strings.Split(line, "x")
		var dims []int
		for _, part := range parts {
			num, err := strconv.ParseInt(strings.TrimSpace(part), 10 ,32)
			if err != nil {
				fmt.Println("Failed to parse line!")
			}
			dims = append(dims, int(num))
		}
		result = append(result, dims)
	}
	return result;
}

func wrappingPapers(intput [][]int) int {
	total := 0
	for _, dims := range intput {
		wrapping := 2 * dims[0] * dims[1] + 2 * dims[1] * dims[2] + 2 * dims[0] * dims[2]
		slices.Sort(dims)
		slack := dims[0] * dims[1]
		total += wrapping + slack
	}
	return total
}

func ribbons(input [][]int) int {
	total := 0
	for _, dims := range input {
		wrapping := dims[0] * dims[1] * dims[2]
		slices.Sort(dims)
		slack := 2 * dims[0] + 2 * dims[1]
		total += wrapping + slack
	}
	return total
}

func main() {
	filename := "2.txt"
	input := parseFile(filename)
	ans1 := wrappingPapers(input)
	ans2 := ribbons(input)
	fmt.Println("Answer for part1:", ans1)
	fmt.Println("Answer for part2:", ans2)
}